content/source/docs/glossary.html.md

```diff
@@ -0,0 +1,940 @@
+
+## Sentinel
+
+[sentinel]: glossary.html#sentinel
+
+-> Terraform Enterprise
+
+A language and runtime for managing policy as code. Allows you to define rules around operations in TFE.
+
+- [TFE docs: Sentinel](/docs/enterprise/sentinel/index.html)
+
+## Site Admin
+
+[site admin]: glossary.html#site-admin
+[site admins]: glossary.html#site-admin
+
+-> Terraform Enterprise
+
+An administrator of a [Private Terraform Enterprise][] instance, who has access to application settings that affect all [organizations][] using the instance.
+
+- [PTFE docs: Administration](/docs/enterprise/private/admin/index.html)
+
+## Speculative Plan
+
+[speculative plan]: glossary.html#speculative-plan
+[speculative plans]: glossary.html#speculative-plan
+
+-> Terraform Enterprise
+
+A [run][] that is only intended to show possible changes to infrastructure, when using a given [config version][] and set of [variables][]. Speculative plans can never be [applied][apply-v], and are usually started as a result of [pull requests][] to a [workspace][]'s repo, or by running `terraform plan` on the command line with [remote operations][] configured.
+
+## SSH Key
+
+[ssh key]: glossary.html#ssh-key
+[ssh keys]: glossary.html#ssh-key
+
+A type of access credential based on public key cryptography, used to log into servers.
+
+TFE uses SSH private keys for two kinds of operations:
+
+- Downloading private Terraform [modules][] with [Git][]-based sources during a Terraform run. Keys for downloading modules are assigned per-workspace.
+- Bringing content from a connected [VCS provider][] into TFE, usually when downloading a Terraform [configuration][] for a [workspace][] or importing a module into the [private module registry][]. Only some VCS providers require an SSH key, but others can optionally use SSH instead of the provider's normal API.
+
+- [Wikipedia: SSH](https://en.wikipedia.org/wiki/Secure_Shell)
+- [TFE docs: SSH Keys for Cloning Modules](/docs/enterprise/workspaces/ssh-keys.html)
+
+## State
+
+[state]: glossary.html#state
+
+Terraform's cached information about your managed infrastructure and [configuration][]. This state is used to persistently map the same real world [resources][] to your configuration from run-to-run, keep track of metadata, and improve performance for large infrastructures.
+
+Without state, Terraform has no way to identify the real resources it created during previous runs. Thus, when multiple people are collaborating on shared infrastructure, it's important to store state in a shared location, like a free Terraform Enterprise organization.
+
+- [Terraform docs: State](/docs/state/index.html)
+
+## State Version
+
+[state version]: glossary.html#state-version
+[state versions]: glossary.html#state-version
+
+-> Terraform Enterprise
+
+A snapshot of your infrastructure's [state][] at a point in time. Can be manually uploaded to TFE or created as the result of an [apply][].
```

Comment:
It's funny that you used the word "snapshot" in the prose here because I was going to point out that Terraform CLI calls these "state _snapshots_", as opposed to versions. I tend to think that "version" connotes an intentionally-created/curated snapshot (like a software release, or a git tag) while this is literally just a copy of the state as it existed at some arbitrary point in time, as you wrote.

The comment I left in reply to @armchairlinguist below also relied on the concept of "snapshot", and I think that terminology makes it easier to explain Terraform CLI's behavior: it will, at various points, take a snapshot of the current state and write it to the backend. The backend is then responsible for storing _at least_ the latest snapshot, and optionally historical snapshots.

Perhaps this idea of "state version" is too far gone at this point to change, but I can't help but feel it gives the concept more levity than it deserves.

Reply:
I think the "snapshot" word came from my initial write-up, which stemmed from our conversations a while back. This may or may not be a great spot to start correcting people's understanding of our state handling, in spite of our "version" wording. Seems like it would need clarified here _and_ in other docs at some point if we wanted to go ahead and do that. Otherwise maybe leave it out of this doc for now?

Reply:
@apparentlymart I hear you about "version" implying more _intent_ than it actually involves. But since the UI and API today both use the term, I'm sticking with it in the glossary for now. But I'll revise that definition to carry a little more context — thank you for the explanations!